diff --git a/BUILD.gn b/../patchs/build_modified.gn
index 38739d110..9af0aca61 100644
--- a/BUILD.gn
+++ b/../patchs/build_modified.gn
@@ -23,6 +23,10 @@ config("pdfium_common_config") {
     "PNG_USE_READ_MACROS",
   ]
 
+  if (is_ios) {
+    defines += [ "MEMORY_TOOL_REPLACES_ALLOCATOR" ]
+  }
+
   if (!use_system_libopenjpeg2) {
     defines += [ "OPJ_STATIC" ]
   }
@@ -223,57 +227,6 @@ group("pdfium_unittest_deps") {
   ]
 }
 
-test("pdfium_unittests") {
-  testonly = true
-  sources = [
-    "testing/unit_test_main.cpp",
-  ]
-  deps = [
-    "core/fdrm:unittests",
-    "core/fpdfapi/edit:unittests",
-    "core/fpdfapi/font:unittests",
-    "core/fpdfapi/page:unittests",
-    "core/fpdfapi/parser:unittests",
-    "core/fpdfapi/render:unittests",
-    "core/fpdfdoc:unittests",
-    "core/fpdftext:unittests",
-    "core/fxcodec:unittests",
-    "core/fxcrt",
-    "core/fxcrt:unittests",
-    "core/fxge:unittests",
-    "fpdfsdk:unittests",
-    "testing:test_support",
-    "testing:unit_test_support",
-    "//testing/gmock",
-    "//testing/gtest",
-  ]
-  configs += [ ":pdfium_core_config" ]
-
-  if (is_android) {
-    use_raw_android_executable = true
-  }
-
-  if (pdf_enable_v8) {
-    configs += [ "//v8:external_startup_data" ]
-    deps += [
-      "fxjs:unittests",
-      "//v8",
-    ]
-  }
-
-  if (pdf_enable_xfa) {
-    deps += [
-      "fxbarcode:unittests",
-      "xfa/fde:unittests",
-      "xfa/fgas:unittests",
-      "xfa/fgas/layout:unittests",
-      "xfa/fxfa:unittests",
-      "xfa/fxfa/fm2js:unittests",
-      "xfa/fxfa/parser:unittests",
-    ]
-  }
-}
-
 group("pdfium_embeddertest_deps") {
   testonly = true
   public_deps = [
@@ -293,51 +246,6 @@ group("pdfium_embeddertest_deps") {
   ]
 }
 
-test("pdfium_embeddertests") {
-  testonly = true
-  sources = [
-    "testing/embedder_test_main.cpp",
-  ]
-  deps = [
-    "core/fpdfapi/edit:embeddertests",
-    "core/fpdfapi/parser:embeddertests",
-    "core/fpdfapi/render:embeddertests",
-    "core/fxcodec:embeddertests",
-    "core/fxcrt",
-    "core/fxge:embeddertests",
-    "fpdfsdk:embeddertests",
-    "fpdfsdk/pwl:embeddertests",
-    "testing:test_support",
-    "testing/image_diff",
-    "//testing/gmock",
-    "//testing/gtest",
-  ]
-  include_dirs = [ "testing/gmock/include" ]
-  configs += [ ":pdfium_core_config" ]
-
-  if (is_android) {
-    ignore_all_data_deps = true
-    use_raw_android_executable = true
-  }
-
-  if (pdf_enable_v8) {
-    deps += [
-      "fxjs:embeddertests",
-      "//v8",
-    ]
-    configs += [ "//v8:external_startup_data" ]
-  }
-
-  if (pdf_enable_xfa) {
-    deps += [
-      "fpdfsdk/fpdfxfa:embeddertests",
-      "xfa/fwl:embeddertests",
-      "xfa/fxfa/layout:embeddertests",
-      "xfa/fxfa/parser:embeddertests",
-    ]
-  }
-}
-
 executable("pdfium_diff") {
   testonly = true
   sources = [
@@ -374,8 +282,6 @@ group("pdfium_all") {
   testonly = true
   deps = [
     ":pdfium_diff",
-    ":pdfium_embeddertests",
-    ":pdfium_unittests",
   ]
   if (pdf_is_standalone) {
     deps += [
