--- find_sdk.py	2019-11-18 01:50:29.000000000 -0300
+++ ../../../patchs/find_sdk_modified.py	2019-11-18 01:50:08.000000000 -0300
@@ -42,7 +42,7 @@
 
 def parse_version(version_str):
   """'10.6' => [10, 6]"""
-  return map(int, re.findall(r'(\d+)', version_str))
+  return list(map(int, re.findall(r'(\d+)', version_str)))
 
 
 def main():
@@ -80,7 +80,7 @@
       print(out, file=sys.stderr)
       print(err, file=sys.stderr)
       raise Exception('Error %d running xcode-select' % job.returncode)
-    dev_dir = out.rstrip()
+    dev_dir = out.rstrip().decode('utf-8')
   sdk_dir = os.path.join(
       dev_dir, 'Platforms/MacOSX.platform/Developer/SDKs')
 
