--- cfx_fontmapper.cpp	2019-11-26 15:26:48.000000000 -0300
+++ ../../../patchs/cfx_fontmapper_modified.cpp	2019-11-26 15:26:22.000000000 -0300
@@ -194,16 +194,16 @@
 }
 
 ByteString ParseStyle(const char* pStyle, int iLen, int iIndex) {
-  std::ostringstream buf;
+  std::vector<char> buf;
   if (!iLen || iLen <= iIndex)
-    return ByteString(buf);
+    return ByteString(buf.data());
   while (iIndex < iLen) {
     if (pStyle[iIndex] == ',')
       break;
-    buf << pStyle[iIndex];
+    buf.push_back(pStyle[iIndex]);
     ++iIndex;
   }
-  return ByteString(buf);
+  return ByteString(buf.data());
 }
 
 const struct FX_FontStyle {
