diff --git a/toolchain.gni b/../../../patchs/toolchain_modified.gni
index 3edc97629..a56e29e2e 100644
--- a/toolchain.gni
+++ b/../../../patchs/toolchain_modified.gni
@@ -44,8 +44,8 @@ declare_args() {
 # Check target_os here instead of is_ios as this file is loaded for secondary
 # toolchain (host toolchain in particular) but the argument is the same for
 # all toolchains.
-assert(!use_xcode_clang || target_os == "ios",
-       "Using Xcode's clang is only supported in iOS builds")
+#assert(!use_xcode_clang || target_os == "ios",
+#       "Using Xcode's clang is only supported in iOS builds")
 
 # Extension for shared library files (including leading dot).
 if (is_mac || is_ios) {
